### PR TITLE
EZP-29143 run test features on Demo version without changing feature files

### PR DESCRIPTION
**Pull request by m-tyrala**
| Question      | Answer
| ------------- | ---
| Tickets       | [EZP-29143](https://jira.ez.no/browse/EZP-29143)
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes/no
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Testing reusability of admin-ui tests with demo environments.

#### Checklist:
- [ ] Coding standards (`$ composer fix-cs`)
- [ ] Ready for Code Review

### CODE REVIEW

File: src/lib/Behat/BusinessContext/ContentUpdateContext.php [hunk 1]
```diff
@@ -53,4 +54,24 @@ public function iClickCloseButton(): void
         $updateItemPage = PageObjectFactory::createPage($this->utilityContext, ContentUpdateItemPage::PAGE_NAME, '');
         $updateItemPage->contentUpdateForm->closeUpdateForm();
     }
+
+    /**
+     * @When I set article intro :intro
+     */
+    public function iSetArticleFields(string $intro): void
```
Comment (mnocon) [inline src/lib/Behat/BusinessContext/ContentUpdateContext.php]: Please rename this function

File: features/ContentCreation.feature [hunk 2]
```diff
@@ -33,7 +33,7 @@ Scenario: Content can be previewed during creation
     And I set content fields
       | label | value               |
       | Title | Test Article2       |
-      | Intro | Test article2 intro |
+    And I set article intro "Test article2 intro"
```
Comment (mnocon) [inline features/ContentCreation.feature]: I'm not sure about intro, I'd go with something like "article main content field" (but shorter :D )

File: src/lib/Behat/Helper/PlatformEnvironmentConstants.php [hunk 3]
```diff
@@ -16,6 +16,9 @@ public function __construct()
         $this->values = [
             'ROOT_CONTENT_NAME' => 'eZ Platform',
             'ROOT_CONTENT_TYPE' => 'Folder',
+            'ARTICLE_FIELD_NAME' => 'Intro',
```
Comment (mnocon) [inline src/lib/Behat/Helper/PlatformEnvironmentConstants.php]: Needs a better name IMHO.

File: src/lib/Behat/PageElement/ElementFactory.php [hunk 4]
```diff
@@ -43,8 +43,12 @@ private static function getFactory(int $installType): ElementFactory
         switch ($installType) {
             case InstallType::PLATFORM:
                 return new PlatformElementFactory();
+            case InstallType::PLATFORM_DEMO:
```
Comment (mnocon) [inline src/lib/Behat/PageElement/ElementFactory.php]: I like the form:
```
            case InstallType::PLATFORM:
            case InstallType::PLATFORM_DEMO:
                 return new PlatformElementFactory();
             case InstallType::ENTERPRISE:
             case InstallType::ENTERPRISE_DEMO:
                 return new EnterpriseElementFactory();
```
better, for me it clearly shows that we're using only two factories.

**Comment by m-tyrala** [comment]
@lserwatka ready to merge

**Pull request merged**